orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/cluster/AbstractClusterWideClouddriverOperationStage.groovy

```diff
@@ -38,6 +46,23 @@ abstract class AbstractClusterWideClouddriverOperationStage implements StageDefi
     return taskClassSimpleName
   }
 
+  public List<String> getLockNames(Stage stage) {
+    final ClusterSelection cluster = stage.mapTo(AbstractClusterWideClouddriverTask.ClusterSelection.class)
+    return getLocations(stage).collect { LockNameHelper.buildClusterLockName(cluster.cloudProvider, cluster.credentials, cluster.cluster, it.value) }
+  }
+
+  static List<Location> getLocations(Stage stage) {
+   List<Location> locations =
+    stage.context.namespaces ? stage.context.namespaces.collect { new Location(type: Location.Type.NAMESPACE, value: it) } :
+     stage.context.regions ? stage.context.regions.collect { new Location(type: Location.Type.REGION, value: it) } :
+       stage.context.zones ? stage.context.zones.collect { new Location(type: Location.Type.ZONE, value: it) } :
+         stage.context.namespace ? [new Location(type: Location.Type.NAMESPACE, value: stage.context.namespace)] :
+           stage.context.region ? [new Location(type: Location.Type.REGION, value: stage.context.region)] :
+             []
```

Comment:
I wish we were checking for `stage.context.zone` here

Reply:
seems like the responsible thing to do